client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java

```diff
@@ -0,0 +1,306 @@
+                        packetModel.ProgressBarVisible(false);
+                        break;
+                    case UPLOAD_ALREADY_COMPLETED:
+                        packetModel.setProgress(PROGRESS_UPLOAD_COMPLETED);
+                        packetModel.ProgressBarVisible(false);
+                        Toast.makeText(this, "Packet already uploaded", Toast.LENGTH_LONG).show();
+                        break;
+                    case SYNC_FAILED:
+                    case UPLOAD_FAILED:
+                        packetModel.setProgress(PROGRESS_RESET);
+                        packetModel.ProgressBarVisible(false);
+                        break;
+                }
+                adapter.notifyDataSetChanged();
+            });
+        } catch (Exception ex) {
+            Log.e(TAG, "syncAndUploadPacket: ", ex);
+        }
+    }
+
+    private class ActionCallback implements ActionMode.Callback {
+        @Override
+        public boolean onCreateActionMode(ActionMode mode, Menu menu) {
+            PacketListActivityHelper.toggleStatusBarColor(PacketListActivity.this, R.color.blue);
+            mode.getMenuInflater().inflate(R.menu.upload_menu, menu);
+            return true;
+        }
+
+        @Override
+        public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
+            return false;
+        }
+
+        @Override
+        public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
+            switch (item.getItemId()) {
+                case R.id.upload:
+                    syncAndUploadSelectedPacket();
+                    mode.finish();
+                    Toast.makeText(getApplicationContext(), R.string.packets_upload_scheduled, Toast.LENGTH_SHORT).show();
+                    return true;
+                case R.id.select_all:
+                    adapter.clearSelection();
+                    for (int position = 0; position < adapter.getItemCount(); position++) {
+                        toggleSelection(position);
+                    }
+                    return true;
+            }
+            return false;
+        }
+
+        @Override
+        public void onDestroyActionMode(ActionMode mode) {
+            adapter.clearSelection();
+            actionMode = null;
+            PacketListActivityHelper.toggleStatusBarColor(PacketListActivity.this, R.color.primaryColor);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.packet_list_menu, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+        } else {
+            Intent i = new Intent(PacketListActivity.this, SettingsActivity.class);
+            startActivity(i);
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private boolean checkNetwork() {
```

Comment:
This should be part of some Util class as we may need to reuse this method in other activities.

Reply:
Network Utility methods added in NetworkUtil class.